update-motd.d/sysinfo.py

```diff
@@ -3,109 +3,166 @@
+        seen = ""
+        for a in l.split():
+            if seen == "src":
+                return a
+            seen = a
+    return None
+
+
+def default_dev():
+    """find the device where our default route is"""
+    for l in open("/proc/net/route").readlines():
+        a = l.split()
+        if a[1] == "00000000":
+            return a[0]
+    return None
+
+
+def get_users():
+    """Get the users connected on this machine."""
+    # Run the who command and capture its output
+    who_output_list = subprocess.check_output(["who"]).decode("utf-8").split("\n")
+    # Create a set to store unique usernames
+    unique_users = set()
+
+    # Iterate through each line and extract usernames
+    for line in who_output_list:
+        line_list = line.split()
+        if bool(line_list):
+            unique_users.add(line_list[0])
+
+    # Join the unique usernames into a single string separated by commas
+    result = ", ".join(unique_users)
+
+    return result
+
 
 def proc_meminfo():
-    """ Get memory usage informations
-    """
+    """Get memory usage informations"""
     items = {}
-    for line in open('/proc/meminfo', encoding="ASCII").readlines():  # pylint: disable-msg=R1732
+    for line in open(
+        "/proc/meminfo", encoding="ASCII"
+    ).readlines():  # pylint: disable-msg=R1732
         array = line.split()
         items[array[0]] = int(array[1])
     return items
 
-def proc_mount():
-    """ Get disks space usage
-    """
-    items = {}
-    for mount in open('/proc/mounts', encoding="ASCII").readlines():  # pylint: disable-msg=R1732
-        array = mount.split()
-        if array[0].find('/dev/') == 0:
-            l_statfs = os.statvfs(array[1])
-            perc = 100 - 100. * l_statfs.f_bavail / l_statfs.f_blocks \
-                    if l_statfs.f_blocks != 0 else 100
-            g_b = l_statfs.f_bsize * l_statfs.f_blocks / 1024. / 1024 / 1024
-            items[array[1]] = f"{perc:5.1f}% of {g_b:.2f}GB"
-    return items
 
-def inode_proc_mount():
-    """ Get disks inode usage
-    """
-    items = {}
-    for mount in open('/proc/mounts', encoding="ASCII").readlines():  # pylint: disable-msg=R1732
-        array = mount.split()
-        if array[0].find('/dev/') == 0:
-            l_statfs = os.statvfs(array[1])
-            perc = 100 - 100. * l_statfs.f_ffree / l_statfs.f_files \
-                    if l_statfs.f_files != 0 else 100
-            i_total = l_statfs.f_files
-            items[array[1]] = f"{perc:5.1f}% of {i_total}"
-    return items
+def get_filesystems():
+    """Get the real filesystem information for all entries in /etc/fstab."""
+    # Only using fstab values to filter the mess that can be containerisation or bind mounts.
+    filesystems = json.loads(
+        subprocess.check_output(
+            [
+                "findmnt",
+                "--noheading",
+                "--real",
+                "--uniq",
+                "--fstab",
```

Comment:
With `--fstab`, you won’t show disks mounted with [systemd mount unit](https://www.freedesktop.org/software/systemd/man/latest/systemd.mount.html).